cmd/datactl/app/sources/add/add_lmt.go

```diff
@@ -0,0 +1,206 @@
+package add
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+
+	"emperror.dev/errors"
+	"github.com/manifoldco/promptui"
+	datactlapi "github.com/redhat-marketplace/datactl/pkg/datactl/api"
+	"github.com/redhat-marketplace/datactl/pkg/datactl/config"
+	"github.com/redhat-marketplace/datactl/pkg/printers/output"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	configInitLongLmt = templates.LongDesc(i18n.T(`
+		Adds a source of the licence metric tool to the config file with details about the source & source type.
+		The command will attempt to add the source name & source type of the specified source in config.`))
+
+	configInitExampleLmt = templates.Examples(i18n.T(`
+		# Initialize the source, using the host, port and token.
+		{{ .cmd }} sources add --host --port --token
+`))
+)
+
+func NewCmdAddLmt(rhmFlags *config.ConfigFlags, f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := addLmtOptions{
+		rhmConfigFlags: rhmFlags,
+		IOStreams:      streams,
+		testFlag:       true,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "lmt",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Initializes the config for LMT source details"),
+		Long:                  output.ReplaceCommandStrings(configInitLongLmt),
+		Example:               output.ReplaceCommandStrings(configInitExampleLmt),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.LmtSourceType, "LmtSourceType", "", i18n.T("source type of the licence metric tool"))
```

Comment:
This flag isn't necessary. We know what type is because it's the ilmt source.

Reply:
flag we have removed but still we are in impression that changes needs to be made configurable for all licensing tools such as Bigfix, Flexera, Tanium other than ILMT.

Reply:
Each source will have it's own add command `add_bigfix.go`. This means that having a source type flag isn't necessary as each command will look like:

```sh
datactl sources add bigfix 
datactl sources add ilmt
datactl sources add dataservice
```

As a note, it's totally fine to create generic functions or a struct to use between sources that have similar endpoints.

Reply:
ok noted